### DISCUSSION TITLE
why no admin endpoint to get session from `session.id`?

### DISCUSSION THREAD

**Discussion by hugotiburtino** [ory/kratos#2704] [Q&A] [answered]
I suppose the answer is: you have security concerns, but I just wanted have it asked to be sure.

I can tell my pain: we have a public graphql API (i.e. cors *) and, when frontend app calls the graphql endpoint, it can't send cookies unless it sets `credentials: true`, what leads to the problem of not being allowed to send to a backend with cors set to *. The authentication was made in frontend (it makes much more sense to do it there), so both api and frontend don't have access to the session token, but frontend could send the session id. The API could use this session id to authenticate the user (you may say, that is more the job of hydra and we could that way, but then we would have an extra layer of configuration...).

**Comment by aeneasr** [comment, marked as answer]
Setting CORS to catch all is bad practice (it’s like disabling CORS protections) so it is good to restrict it in any case!

Regarding your question, we might at some point introduce an API which allows you to exchange the session for a short lived JWT (eg 1 minute). In the meanwhile you could transform the session token to a JWT in your edge as well using Ory Oathkeeper for example!
